src/components/negotiations-panel.tsx

```diff
@@ -61,6 +63,10 @@ export default function NegotiationsPanel(props: NegotiationsPanelProps): JSX.El
     // TODO: spotlight
   }, [spotlight]);
 
+  let curLevelName = "";
+  if (mapFloorLayerSorted)
+    curLevelName = mapFloorLayerSorted[0];
+
```

Comment:
What do you think about doing something like this?

``` ts
  const curLevelName = React.useMemo(() => {
    if (!mapFloorLayerSorted) {
      return ''; // or maybe you can return null
    }
    return mapFloorLayerSorted[0];
  }, [mapFloorLayerSorted]);
```

The idea here is to memoize the result in case the input is the same. In this case, this assignation (curLevelName=return mapFloorLayerSorted[0]) will not occur again until the value mapFloorLayerSorted changes.

Reply:
is it even worth it for to memoize a string that can be easily evaluated? my understanding of the side effects is that it increases memory usage because it keeps track of dependencies. i understand not re-rendering components, but this seems not worth it. correct me if I'm wrong.

also there's an incoming PR #105 that memoizes the whole panel, let the effort go there

Reply:
> is it even worth it for to memoize a string that can be easily evaluated? my understanding of the side effects is that it increases memory usage because it keeps track of dependencies.

I totally agree with you in normal situations, but we should check how are the array `mapFloorLayerSorted` changes affecting to the components re-render. We can do that using React-Profiler. https://reactjs.org/blog/2018/09/10/introducing-the-react-profiler.html.

Sure we can take it a look at that in the FUP PR.